climakitae/explore/warming.py

```diff
@@ -697,101 +722,163 @@ def warming_levels_select(self):
-            if not any_single_dims:
-                all_plots = all_plot_data.hvplot.image(**plot_image_kwargs).cols(4)
-            else:
-                # Aggregate all data to just the `all_sims` dimension. This will average the data across all dimensions, which may not necessarily be desired for calculations with 'Max' variables, if you are for instance looking for a 'max of maxes'.
-                all_plot_data = all_plot_data.mean(
-                    dim=[dim for dim in all_plot_data.dims if dim != "all_sims"]
+        # If allllll the simulations don't reach the warming level, print a message
+        if data_to_plot.isnull().all().item():
+            warm_level_dict[warmlevel] = pn.widgets.StaticText(
+                value=("<b>No simulations reach this warming level</b>"),
+                width=300,
+                style={
+                    "border": "1.2px red solid",
+                    "padding": "5px",
+                    "border-radius": "4px",
+                    "font-size": "13px",
+                },
+            )
+            continue
+
+        # If some of the simulations reach the warming level, but not ALL, remove that simulation
+        # This is so an empty plot isn't generated
+        for sim in data_to_plot.simulation.values:
+            if data_to_plot.sel(simulation=sim).isnull().all().item() == True:
+                data_to_plot = data_to_plot.where(
+                    data_to_plot["simulation"] != sim, drop=True
                 )
 
-                # Remove SSP descriptions
-                all_plot_data["all_sims"] = [
-                    sim_name.item().split("--")[0].strip()
-                    for sim_name in all_plot_data.all_sims
-                ]
+        # Get min and max to use for colorbar
+        vmin, vmax, sopt = compute_vmin_vmax(data_to_plot.min(), data_to_plot.max())
 
-                if wl_viz.wl_params.downscaling_method == "Dynamical":
-                    # Creating barh plot since there's only max 8 WRF simulations, so each bar and label is still legible
-                    all_plots = all_plot_data.hvplot.barh(
-                        x="all_sims", xlabel="Simulation", ylabel=f"{units}"
-                    ).opts(multi_level=False, show_legend=False)
+        # Get cmap
+        cmap = _get_cmap(wl_viz.wl_params.variable, wl_viz.variable_descriptions, vmin)
 
-                else:
-                    # Creating histogram since all simulations are too many to put on a bar plot
-                    all_plots = all_plot_data.hvplot.hist()
+        # If there are less than or equal to four simulations, make postage stamps
+        if len(data_to_plot.simulation.values) <= 4:
 
-            try:
-                all_plots.opts(
-                    title=wl_viz.wl_params.variable
+            # if there's only one data point, make a scatter plot
+            if len(data_to_plot.x.values) == 1 and len(data_to_plot.y.values) == 1:
```

Comment:
This breaks in `warming_levels_approach` when `data_to_plot` comes from a Statistical downscaling method.

Reply:
Great thanks!